com/metacube/shoppingcart/facade/ProductFacade.java

```diff
@@ -0,0 +1,79 @@
+package com.metacube.shoppingcart.facade;
+
+import java.util.List;
+
+import com.metacube.shoppingcart.dao.ProductDao;
+import com.metacube.shoppingcart.entity.Cart;
+import com.metacube.shoppingcart.entity.Product;
+
+public class ProductFacade {
+	
+	private static ProductFacade facadeObject;
+	private ProductDao productDao = ProductDao.getInstance();
+	private ProductFacade(){}
+	
+	public static ProductFacade getInstance(){
+		if(facadeObject == null){
+			facadeObject = new ProductFacade(); 
+		}
+		return facadeObject;
+	}
+	
+	public List<Product> getProductList(){
+		return productDao.getProductList();	 
+	}
+	
+	private boolean checkProductAvailability(Product product){
+		List<Product> productList = getProductList();
+		return productList.contains(product.getProductName());
```

Comment:
You do not need this method. getProductById(int productId) method is doing the same thing. And more better you no need such type of method. You can simply create a map<Id, product>. This map will make your code short and easy.